utils/node_local_test.py

```diff
@@ -934,6 +946,122 @@ def make_pool(daos):
 
     return get_pool_list()
 
+def needs_dfuse(method):
+    """Decorator function for starting dfuse under posix_tests class"""
+    @functools.wraps(method)
+    def _helper(self):
+        self.dfuse = DFuse(self.server,
+                           self.conf,
+                           pool=self.pool,
+                           container=self.container)
+        self.dfuse.start(v_hint=method.__name__)
+        rc = method(self)
+        if self.dfuse.stop():
+            self.fatal_errors = True
+        return rc
+    return _helper
+
+
+class posix_tests():
+    """Class for adding standalone unit tests"""
+
+    def __init__(self, server, conf, pool=None, container=None):
+        self.server = server
+        self.conf = conf
+        self.pool = pool
+        self.container = container
+        self.dfuse = None
+        self.fatal_errors = False
+
+    # pylint: disable=no-self-use
+    def fail(self):
+        """Mark a test method as failed"""
+        raise NLTestFail
+
+    @needs_dfuse
+    def test_open_replaced(self):
+        """Test that fstat works on file clobbered by rename"""
+        fname = os.path.join(self.dfuse.dir, 'unlinked')
+        newfile = os.path.join(self.dfuse.dir, 'unlinked2')
+        ofd = open(fname, 'w')
+        nfd = open(newfile, 'w')
+        nfd.write('hello')
+        nfd.close()
+        print(os.fstat(ofd.fileno()))
+        os.rename(newfile, fname)
+        # This should fail, because the file has been deleted.
+        try:
+            print(os.fstat(ofd.fileno()))
+            self.fail()
+        except FileNotFoundError:
+            print('Failed to fstat() renamed file')
+        ofd.close()
+
+    @needs_dfuse
+    def test_open_rename(self):
+        """Check that fstat() on renamed files works as expected"""
+        fname = os.path.join(self.dfuse.dir, 'unlinked')
+        newfile = os.path.join(self.dfuse.dir, 'unlinked2')
+        ofd = open(fname, 'w')
+        pre = os.fstat(ofd.fileno())
+        print(pre)
+        os.rename(fname, newfile)
+        try:
+            post = os.fstat(ofd.fileno())
+            print(post)
+            self.fail()
+        except FileNotFoundError:
+            print('Failed to fstat() renamed file')
+        os.stat(newfile)
+        post = os.fstat(ofd.fileno())
+        print(post)
+        assert pre.st_ino == post.st_ino
+        ofd.close()
+
+    @needs_dfuse
+    def test_open_unlinked(self):
+        """Test that fstat works on unlinked file"""
+        fname = os.path.join(self.dfuse.dir, 'unlinked')
+        ofd = open(fname, 'w')
+        print(os.fstat(ofd.fileno()))
+        os.unlink(fname)
+        try:
+            print(os.fstat(ofd.fileno()))
+            self.fail()
+        except FileNotFoundError:
+            print('Failed to fstat() renamed file')
```

Comment:
suggest change the string to "Failed to fstat() unlinked file" ("renamed"->"unlinked").

Reply:
Done, thanks.